x/stakeibc/keeper/delegation.go

```diff
@@ -93,53 +111,56 @@ func (k Keeper) StakeExistingDepositsOnHostZones(ctx sdk.Context, epochNumber ui
 	stakeDepositRecords := utils.FilterDepositRecords(depositRecords, func(record recordstypes.DepositRecord) (condition bool) {
 		isStakeRecord := record.Status == recordstypes.DepositRecord_DELEGATION_QUEUE
 		isBeforeCurrentEpoch := record.DepositEpochNumber < epochNumber
-		return isStakeRecord && isBeforeCurrentEpoch
+		isNotInProgress := record.DelegationTxsInProgress == 0
+		return isStakeRecord && isBeforeCurrentEpoch && isNotInProgress
 	})
 
 	if len(stakeDepositRecords) == 0 {
 		k.Logger(ctx).Info("No deposit records in state DELEGATION_QUEUE")
 		return
 	}
 
-	// limit the number of staking deposits to process per epoch
-	maxDepositRecordsToStake := utils.Min(len(stakeDepositRecords), cast.ToInt(k.GetParam(ctx, types.KeyMaxStakeICACallsPerEpoch)))
```

Comment:
Nit: should we remove the type?

Reply:
eh - there's a lot going on already so let's hold off for now. I think we can make it a separate ticket to cleanup our params cause I think we have a lot that aren't actually used

Reply:
sgtm